oneclick.py

```diff
@@ -52,28 +55,44 @@
 class OneClickAPI(object):
 
     API_VERSION = "v1"
+    PRODUCTION_BASE_URL = "https://api.oneclickdigital.com/"
+    QA_BASE_URL = "https://api.oneclickdigital.us/"
+    
+    # Map simple nicknames to server URLs.
+    SERVER_NICKNAMES = {
+        "production" : PRODUCTION_BASE_URL,
+        "qa" : QA_BASE_URL,
+    }
+
     DATE_FORMAT = "%Y-%m-%d" #ex: 2013-12-27
 
     # a complete response returns the json structure with more data fields than a basic response does
     RESPONSE_VERBOSITY = {0:'basic', 1:'compact', 2:'complete', 3:'extended', 4:'hypermedia'}
-
+   
     log = logging.getLogger("OneClick API")
 
-    def __init__(self, _db, library_id=None, username=None, password=None, 
-        remote_stage=None, base_url=None, basic_token=None, 
-        ebook_loan_length=None, eaudio_loan_length=None):
+    def __init__(self, _db, collection):
         self._db = _db
-            
-        self.library_id = library_id
-        self.username = username
-        self.password = password
-        self.remote_stage = remote_stage
-        self.base_url = base_url or ''
-        self.base_url = self.base_url + self.API_VERSION
-        self.token = basic_token
+
+        if collection.protocol != collection.ONE_CLICK:
+            raise ValueError(
+                "Collection protocol is %s, but passed into OneClickAPI!" %
+                collection.protocol
+            )
+        
+        self.library_id = collection.external_account_id.encode("utf8")
+        self.username = collection.username.encode("utf8")
```

Comment:
Fyi: I'm looking at the OneClick code, and it looks like we're not actually using the library username to talk to them.